### PR TITLE
XAMEETINGS-148 : Don't display PARTICIPATING GROUPS field if value is empty #15

### PR DESCRIPTION
**Pull request by CostiChD**

### CODE REVIEW

File: ui/src/main/resources/Meeting/Code/MeetingSheet.xml [hunk 1]
```diff
@@ -236,19 +236,19 @@
       #end
       : $doc.display('participants')
       ## groups
-      ; &lt;label for="Meeting.Code.MeetingClass_0_groups"&gt;$!services.icon.render('group') $services.localization.render('contrib.meeting.field.groups')&lt;/label&gt;##
-      #if ($xcontext.action == 'edit')
+      #if ($xcontext.action == 'edit' || $doc.getValue('groups') != '')
+        ; &lt;label for="Meeting.Code.MeetingClass_0_groups"&gt;$!services.icon.render('group') $services.localization.render('contrib.meeting.field.groups')&lt;/label&gt;##
         (% class="xHint" %)$services.localization.render('')
```
Comment (evalica) [inline ui/src/main/resources/Meeting/Code/MeetingSheet.xml]: This is rendering an empty string, is there any translation associated to that label?

File: ui/src/main/resources/Meeting/Code/MeetingSheet.xml [hunk 2]
```diff
@@ -236,19 +236,19 @@
       #end
       : $doc.display('participants')
       ## groups
-      ; &lt;label for="Meeting.Code.MeetingClass_0_groups"&gt;$!services.icon.render('group') $services.localization.render('contrib.meeting.field.groups')&lt;/label&gt;##
-      #if ($xcontext.action == 'edit')
+      #if ($xcontext.action == 'edit' || $doc.getValue('groups') != '')
```
Comment (acotiuga) [inline ui/src/main/resources/Meeting/Code/MeetingSheet.xml]: The right check here would be `"$!doc.getValue('groups')" != ''`. At the same time, I think you should hide this field all the time, not only in edit mode, if is missing, so there's no reason to check the action.

File: ui/src/main/resources/Meeting/Code/MeetingSheet.xml [hunk 3]
```diff
@@ -236,19 +236,19 @@
...
+        ; &lt;label for="Meeting.Code.MeetingClass_0_groups"&gt;$!services.icon.render('group') $services.localization.render('contrib.meeting.field.groups')&lt;/label&gt;##
```
Comment (acotiuga) [inline ui/src/main/resources/Meeting/Code/MeetingSheet.xml]: The `label` tag is not a self closing tag as per https://www.w3.org/TR/html/sec-forms.html#the-label-element. Please fix this and properly indent it.

File: ui/src/main/resources/Meeting/Code/MeetingSheet.xml [hunk 4]
```diff
@@ -112,7 +112,9 @@
         #set ($title = $doc.getValue('meetingTitle'))
       #end
       #set ($meetingTitleId = 'Meeting.Code.MeetingClass_0_meetingTitle')
-      ; &lt;label for=""&gt;$!services.icon.render('font') $services.localization.render('contrib.meeting.field.meetingTitle')&lt;/label&gt;
+      ; &lt;label for=""&gt;
```
Comment (acotiuga) [inline ui/src/main/resources/Meeting/Code/MeetingSheet.xml]: the value in the `for` attribute is empty and the label won't work properly.

Comment (acotiuga) [reply ui/src/main/resources/Meeting/Code/MeetingSheet.xml]: The code from https://github.com/xwikisas/application-meeting/commit/1a3b93ee04e85af44ef50967d1e9c68cbf8dc8ea wasn't fetched. When you start working on something, make sure you got the latest changes from github.

File: ui/src/main/resources/Meeting/Code/MeetingSheet.xml [hunk 5]
```diff
@@ -112,7 +112,9 @@
         #set ($title = $doc.getValue('meetingTitle'))
       #end
       #set ($meetingTitleId = 'Meeting.Code.MeetingClass_0_meetingTitle')
-      ; &lt;label for=""&gt;$!services.icon.render('font') $services.localization.render('contrib.meeting.field.meetingTitle')&lt;/label&gt;
+      ; &lt;label for="$meetingTitleId"&gt;
```
Comment (acotiuga) [inline ui/src/main/resources/Meeting/Code/MeetingSheet.xml]: Here, you need to add the `for` attribute only in edit mode.

File: ui/src/main/resources/Meeting/Code/MeetingSheet.xml [hunk 6]
```diff
@@ -234,33 +244,38 @@
     (% class="col-xs-12 col-sm-4" %)
     (((
       ## leader
-      ; &lt;label for="Meeting.Code.MeetingClass_0_leader"&gt;
-          $!services.icon.render('user') $services.localization.render('contrib.meeting.field.leader')
-        &lt;/label&gt;
-      : $doc.display('leader')
+      #if ($xcontext.action == 'edit' || "$!doc.getValue('leader')" != '')
```
Comment (evalica) [inline ui/src/main/resources/Meeting/Code/MeetingSheet.xml]: There is a lot of duplication. You could also write a macro with some parameters to handle multiple fields.

**Comment by evalica** [comment]
Also first fix the conflict.

**Pull request merged**